File: 프로그래머스/lv2/이선영/로또의 최고 순위와 최저 순위.java
import java.util.*;

class Solution {
    public int[] solution(int[] lottos, int[] win_nums) {
        int[] answer = new int[2];
        boolean[] flag = new boolean[46];
        
        for(int i=0; i<6; i++){
            flag[win_nums[i]] = true;
        }
        
  
        for(int i=0; i<6; i++){
            int num = lottos[i];

            if(num == 0) {//지워짐
                answer[0]++;
            } else if(flag[num]){ //당첨
                answer[1]++;
                answer[0]++;
            }else{ //낙첨
                continue;
            }
        }
        
        answer[0] = 7-answer[0];
        answer[1] = 7-answer[1];
        if(answer[0] == 7) answer[0] = 6;
        if(answer[1] == 7) answer[1] = 6;
        
        return answer;
    }
}

File: lv1/김유현/로또의 최고순위와 최저순위.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

// 정답 개수 - 랭크 
int ranks[7] = {6, 6, 5, 4, 3, 2, 1};

// 정답 여부
bool winNum[46];

vector<int> solution(vector<int> lottos, vector<int> win_nums) {
    vector<int> answer;
    
    for(int num : win_nums) {
        winNum[num] = true;
    }
    
    int zeroCnt = 0;
    int winCnt = 0;
    
    for(int lotto : lottos) {
        if (lotto == 0) {
            zeroCnt++;
        } else if (winNum[lotto]) {
            winCnt++; 
        }
    }
    
    int maxRank = ranks[zeroCnt + winCnt];
    int minRank = ranks[winCnt];
   
    answer.push_back(maxRank);
    answer.push_back(minRank);
    
    return answer;
}